prestoadmin/__init__.py

```diff
@@ -17,30 +17,89 @@
 
 __version__ = '1.2'  # Make sure to update setup.py too
 
+
 import os
+import sys
+
+main_dir = os.path.dirname(os.path.abspath(os.path.dirname(__file__)))
 
 from fabric.api import env
 
+import fabric_patches
+import mode
+
+from prestoadmin.mode import get_mode, for_mode, MODE_STANDALONE, MODE_SLIDER
+from prestoadmin.util.exception import ConfigFileNotFoundError, \
+    ConfigurationError
+
+
+#
+# Subcommands common to all modes. If anybody knows why fabric_patches is in
+# the list, I'll make a note for the next person.
+#
+__all__ = ['fabric_patches', 'mode']
+
+cfg_mode = None
+try:
+    cfg_mode = get_mode()
+except ConfigFileNotFoundError as e:
+    print >>sys.stderr, \
+        "Select a mode using the subcommand 'mode select <mode>'"
+except ConfigurationError as e:
+    print >>sys.stderr, e.message
+
+
+ADDITIONAL_TASK_MODULES = {
+    MODE_SLIDER: [('yarn_slider.server', 'server'),
+                  ('yarn_slider.slider', 'slider')],
+    MODE_STANDALONE: ['topology',
+                      ('configure_cmds', 'configuration'),
+                      'server',
+                      'connector',
+                      'package',
+                      'collect',
+                      'script',
+                      'plugin']}
+
+
+if cfg_mode is not None:
+    atms = for_mode(cfg_mode, ADDITIONAL_TASK_MODULES)
+    for atm in atms:
+        try:
+            module, subcommand_name = atm
+        except ValueError:
+            module = atm
+            subcommand_name = atm
+
+        __all__.append(subcommand_name)
+
+        components = module.split('.')
+
+        if len(components) == 1:
+            # The simple case...
+            # import <module> as <subcommand_name>
+            globals()[subcommand_name] = __import__(module, globals())
+        else:
+            # The complicated case:
+            # import foo.bar doesn't actually import foo.bar; it imports foo.
+            # This is why, for example, you can't to the following:
+
+            # >>> import os.path
+            # >>> path.join('foo', 'bar', 'baz', 'zot')
```

Comment:
I think you still get the parent module of the leaf module back from __import__ if you pass a fromlist, and since we don't know the name of the module, we'd still have to go fishing in the last parent module's __dict__.

At least that was my interpretation of the docs: https://docs.python.org/2/library/functions.html#__import__